[feature] CMakePresets support for msvc toolset version

### What is your suggestion?

## Suggestion

Would it be possible for the Conan generated CMakePresets.json to include `msvc.toolset` version details (i.e. `v143`, `v142`, etc.) when populating the `toolset.value` field?

This would allow IDEs to know which `vcvarsall.bat` environment to use when building the project.

## Expected Behavior

For example, when cross-compiling with VS2022 x64 to x86, I'd expect to see the conan generated CMakePresets.json look something like:
```diff
"generator": "Ninja",
"cacheVariables": {
    "CMAKE_POLICY_DEFAULT_CMP0091": "NEW",
    "CMAKE_BUILD_TYPE": "RelWithDebInfo"
},
"architecture": {
    "value": "x86",
    "strategy": "external"
},
+"toolset": {
+    "value": "v143,host=x64",
+    "strategy": "external"
+},
"toolchainFile": "C:\\projects\\build\\foo\\RelWithDebInfo\\generators\\conan_toolchain.cmake",
"binaryDir": "C:\\projects\\build\\foo\\RelWithDebInfo
```            

_I can use `tools.cmake.cmaketoolchain:toolset_arch=x64` to add the "host=x64" portion, but I don't have anyway to get `v143` added._

VSCode recognizes this `toolset` / `architecture` combination and correctly chooses VS2022 and calls `vcvarsall.bat amd64_x86` prior to calling CMake.

## Current Behavior

Conan *doesn't* include `toolset` details in the generated CMakePresets.json file, so in order to get IDEs like VSCode to find the right environment, we need to adjust CMakePresets.json (as described above) or worse adjust the CMakeUserPresets.json file as follows:

```diff
{
    "version": 4,
    "vendor": {
        "conan": {}
    },
    "include": [
        "C:\\path\\to\\conan\\CMakePresets.json"
-    ]
+    ],
+    "configurePresets": [
+      {
+        "name": "RelWithDebInfo",
+        "displayName": "RelWithDebInfo",
+        "inherits": "relwithdebinfo",
+        "toolset": {
+          "value": "v143,host=x64",
+          "strategy": "external"
+        }
+    ],
+    "buildPresets": [
+        {
+            "name": "RelWithDebInfo",
+            "displayName": "RelWithDebInfo",
+            "configurePreset": "RelWithDebInfo",
+            "configuration": "RelWithDebInfo"
+        }
+    ]
}
```

Conan already knows *which* vcvarsall to call, and and ensures the correct one gets called in the generated `conanvcvars.bat`.

Conan also appears to know about `msvc.toolset` version information:
https://github.com/conan-io/conan/blob/e33b55a486fd208223524dda49cce02dbe70c214/conans/client/conf/__init__.py#L117
_At least for the legacy toolsets_

And seems to know about the mapping:
https://github.com/conan-io/conan/blob/e0a8ee058bc8dc2d9811b8aeb6999f69aeb78d85/conan/tools/microsoft/visual.py#L53-L60

It seems like Conan could should be able to use the `msvc.toolset` information from the build profile and populate the version field correctly.

## Context
Follow up to https://github.com/conan-io/conan/issues/11623

Some IDEs are capable of targeting multiple toolsets (e.g. VS2015, VS2019, VS2022), and `toolset.value` is the way for the IDE to know which `vcvarsall.bat` to load and use internally while compiling.

> It's the same step that Visual Studio takes for you when the IDE invokes CMake. **Visual Studio parses the active Configure Preset for the host and target architecture specified by toolset and architecture. Visual Studio then sources the specified environment from vcvarsall.bat.** When you build from the Windows command line with Ninja, you'll need to take this step yourself.
_Excerpt from MSDN [Sourcing the environment when building with command-line generators on Windows](https://learn.microsoft.com/en-us/cpp/build/cmake-presets-vs?view=msvc-170#sourcing-the-environment-when-building-with-command-line-generators-on-windows)_

IDE | Support 
----|----------
VSCode | [Already Supported](https://github.com/microsoft/vscode-cmake-tools/search?q=varsForVSInstallation) <br> https://github.com/microsoft/vscode-cmake-tools/pull/2524
QtCreator | [In Progress](https://codereview.qt-project.org/c/qt-creator/qt-creator/+/457588/8/src/plugins/cmakeprojectmanager/cmakeprojectimporter.cpp)<br> [QTCREATOR-BUG 28693](https://bugreports.qt.io/browse/QTCREATORBUG-28693)
CLion | [Feature Request](https://youtrack.jetbrains.com/issue/CPP-31353/Support-external-strategy-of-toolset-and-architecture-fields-in-CMakePresets.json-for-MSVC-compilers.)

Thanks for your support!



### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

Thanks! By adding the following:

```diff
{
    "version": 3,
    "vendor": {
        "conan": {}
    },
    "cmakeMinimumRequired": {
        "major": 3,
        "minor": 15,
        "patch": 0
    },
    "configurePresets": [
        {
            "name": "conan-msvc-193",
            "displayName": "'conan-msvc-193' config",
            "description": "'conan-msvc-193' configure using 'Ninja Multi-Config' generator",
            "generator": "Ninja Multi-Config",
            "cacheVariables": {
                "CMAKE_POLICY_DEFAULT_CMP0091": "NEW",
+               "CMAKE_CXX_COMPILER": "cl.exe"
            },
            "architecture": {
                "value": "x64",
                "strategy": "external"
            },
+           "toolset": {
+               "value": "v143,host=x64",
+               "strategy": "external"
+           },
            "toolchainFile": "D:\\repos\\units\\build\\msvc-193\\generators\\conan_toolchain.cmake",
            "binaryDir": "D:\\repos\\units\\build\\msvc-193"
        }
    ],
    "buildPresets": [
        {
            "name": "conan-msvc-193-release",
            "configurePreset": "conan-msvc-193",
            "configuration": "Release"
        }
    ],
    "testPresets": [
        {
            "name": "conan-msvc-193-release",
            "configurePreset": "conan-msvc-193",
            "configuration": "Release"
        }
    ]
}
```

I was able to make MSVC 193 work somehow. The only problem compared to running from VS Dev prompt is that I get the following warning:

```text
warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc
```

However, when I tried to do the same for MSVC 192 installed under VS2022 it didn't work. I also tried with `-c tools.microsoft.msbuild:vs_version=17` provided to conan install command line with no luck.

```
[driver] Removing C:/Users/Diego/conanws/kk/build/Release/CMakeCache.txt
[driver] Removing C:\Users\Diego\conanws\kk\build\Release\CMakeFiles
[proc] Executing command: C:\ws\cmake\cmake-3.25.3-windows-x86_64\bin\cmake.EXE -DCMAKE_POLICY_DEFAULT_CMP0091=NEW -DCMAKE_BUILD_TYPE=Release -DCMAKE_TOOLCHAIN_FILE=C:/Users/memsharded/conanws/kk/build/Release/generators/conan_toolchain.cmake -SC:/Users/memsharded/conanws/kk -BC:/Users/memsharded/conanws/kk/build/Release -G Ninja
[cmake] -- Using Conan toolchain: C:/Users/memsharded/conanws/kk/build/Release/generators/conan_toolchain.cmake
[cmake] -- Conan toolchain: C++ Standard 14 with extensions OFF
[cmake] -- The CXX compiler identification is unknown
[cmake] CMake Error at CMakeLists.txt:2 (project):
[cmake]   No CMAKE_CXX_COMPILER could be found.
[cmake] 
[cmake]   Tell CMake where to find the compiler by setting either the environment
[cmake]   variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
[cmake]   to the compiler, or to the compiler name if it is in the PATH.
```

--- a/conan/tools/cmake/presets.py
+++ b/conan/tools/cmake/presets.py
@@ -5,7 +5,7 @@ import textwrap
 
 from conan.api.output import ConanOutput, Color
 from conan.tools.cmake.layout import get_build_folder_custom_vars
-from conan.tools.cmake.toolchain.blocks import GenericSystemBlock
+from conan.tools.cmake.toolchain.blocks import GenericSystemBlock, CompilersBlock
 from conan.tools.cmake.utils import is_multi_configuration
 from conan.tools.build import build_jobs
 from conan.tools.microsoft import is_msvc
@@ -157,6 +157,18 @@ class _CMakePresets:
                     "value": arch,
                     "strategy": "external"
                 }
+
+        # Second attempt at https://github.com/conan-io/conan/issues/13136
+        # for cmake-tools to activate environment. Similar to CompilersBlock
+        # Only for cl/clang-cl, other compilers such clang can be breaking (Android)
+        compilers_by_conf = conanfile.conf.get("tools.build:compiler_executables", default={})
+        compiler = conanfile.settings.get_safe("compiler")
+        default_cl = "cl" if compiler == "msvc" and "Ninja" in str(generator) else None
+        for lang in ("c", "cpp"):
+            comp = compilers_by_conf.get(lang, default_cl)
+            if comp and os.path.basename(comp) in ("cl", "cl.exe", "clang-cl", "clang-cl.exe"):
+                lang = {"c": "C", "cpp": "CXX"}[lang]
+                ret["cacheVariables"][f"CMAKE_{lang}_COMPILER"] = comp.replace("\\", "/")
 
         ret["toolchainFile"] = toolchain_file
         if conanfile.build_folder:

--- a/test/integration/toolchains/cmake/test_cmaketoolchain.py
+++ b/test/integration/toolchains/cmake/test_cmaketoolchain.py
@@ -1747,3 +1747,27 @@ def test_declared_stdlib_and_passed():
     client.run('install . -s compiler=sun-cc -s compiler.libcxx=libstdcxx')
     tc = client.load("conan_toolchain.cmake")
     assert 'string(APPEND CONAN_CXX_FLAGS " -library=stdcxx4")' in tc
+
+
+def test_cmake_presets_compiler():
+    profile = textwrap.dedent(r"""
+        [settings]
+        os=Windows
+        arch=x86_64
+        compiler=msvc
+        compiler.version=193
+        compiler.runtime=dynamic
+        [conf]
+        tools.build:compiler_executables={"c": "cl", "cpp": "cl.exe"}
+        """)
+    client = TestClient()
+    conanfile = GenConanfile().with_settings("os", "arch", "compiler")\
+        .with_generator("CMakeToolchain")
+    client.save({"conanfile.py": conanfile,
+                 "profile": profile})
+    client.run("install . -pr:b profile -pr:h profile")
+    presets = json.loads(client.load("CMakePresets.json"))
+    cache_variables = presets["configurePresets"][0]["cacheVariables"]
+    # https://github.com/microsoft/vscode-cmake-tools/blob/a1ceda25ea93fc0060324de15970a8baa69addf6/src/presets/preset.ts#L1095C23-L1095C35
+    assert cache_variables["CMAKE_C_COMPILER"] == "cl"
+    assert cache_variables["CMAKE_CXX_COMPILER"] == "cl.exe"

